query/src/lib.rs

```diff
@@ -50,11 +46,12 @@ pub enum SrcVar {
 }
 
 /// These are the scalar values representable in EDN.
-#[derive(Clone,Debug,Eq,PartialEq)]
+// Floats and doubles only implement PartialEq, but not Eq.
+#[derive(Clone,Debug,PartialEq)]
 pub enum NonIntegerConstant {
     Boolean(bool),
-    BigInteger(BigInt),
-    Float(OrderedFloat<f64>),
+    BigInteger(String),
```

Comment:
No, I mean `query` should depend on `BigInt`. `query` represents parsed queries, and those can contain big integers.

`OrderedFloat` is an implementation detail: `Value` only needs that so that `Value` instances can be put into data structures that require hashing. `BigInt` is an intrinsic part of the data being represented; the only improvements we can make there are to provide useful factories so that *some kinds* of calling code can avoid building their own `BigInt` instances.